### DISCUSSION TITLE
Roadmap for Git PR resource

### DISCUSSION THREAD

**Discussion by deepakmohanakrishnan07** [concourse/concourse#7883] [Help & Support] [answered]
### Git Pull Request Resource

We are running a large and centralized concourse cluster(s) for multiple internal teams to assist their CI needs. The [telia-oss/github-pr-resource](https://github.com/telia-oss/github-pr-resource) is one of our popular and widely adopted concourse resource for managing GHE pull requests. However, we are noticing that the authors/maintainers of that resource have abandoned their contribution to keep it updated. 
It is so fascinating to see that the concourse CI pipelines are leveraging [aoldershaw/github-pr-resource](https://github.com/aoldershaw/github-pr-resource) for managing their PRs. We feel it's a teardown and rebuild of the [telia-oss/github-pr-resource](https://github.com/telia-oss/github-pr-resource) with different behaviors. 

## Roadmap

1. Does concourse maintainers have a plan or interest to include [aoldershaw/github-pr-resource](https://github.com/aoldershaw/github-pr-resource) under the [concourse](https://github.com/concourse) git organization? 
1. Does concourse community have a vision how does a pull request resource should look like? 
1. Does concourse community have a recommendation for managing pull requests?

We would like to get some clarification around these resource before starting our own fork from [telia-oss/github-pr-resource](https://github.com/telia-oss/github-pr-resource) and address few features gaps.

**Comment by taylorsilva** [comment, marked as answer]
1. Background: This resource was made by @aoldershaw as part of the work he did on the `across` step. Aidan no longer works at VMware and isn't active on Concourse anymore. He mostly made it so we could start using the `across` step for what we wanted it for: PRs! 
  Moving forward: In general there are no plans to bring any resources into the concourse org, including this one. If it breaks we'll probably fork and fix it, but we don't expect Aidan to maintain it.
2. Not really, we don't think much about how certain types of resouces should look. I have seen a gitlab resource used internally at some company that's structured like a k8s object. It's source field has a `kind` field that takes values like `PullRequests` `Branches`, `Commits`. I thought it was a neat way to jam a bunch of resources into one image that are all related to one service.
3. Let's see if others chime in

Aidan's resource makes me think of this blog post: https://apenwarr.ca/log/20211229
(His resource is a gift).

I wonder if companies/organizations using Concourse would pay/support devs to build and maintain a library of resources?

**Comment by deepakmohanakrishnan07** [comment]
Thanks for the clarifications @taylorsilva That would be a great idea to recognize and motivate opensource contributors/developers.
